### ISSUE TITLE
There are several problems in deploying proxy

### ISSUE THREAD

**Issue by ktpktr0** [jacksontj/promxy#443]
Deployment architecture: proxy + Victoria metrics + proxy + grafana

1. The proxy wed interface has no targets information

2. The proxy has been connected to remote_ Write, rules is configured, but alert status is not displayed.(Prometheus background display alarm status is normal)

3. Empty result of proxy query(The background query of Prometheus is normal, and the query of Victoria metrics is normal)

Proxy configuration:
```
##
## Regular prometheus configuration
##
global:
  evaluation_interval: 5s
  external_labels:
    source: promxy

# Rule files specifies a list of globs. Rules and alerts are read from
# all matching files.
rule_files:
  - "/conf/rules/linux_system.yml"
  - "/conf/rules/windows_system.yml"
  - "/conf/rules/windows_process.yml"
  - "/conf/rules/docker_container.yml"
  - "/conf/rules/process_exporter.yml"
  # - "second_rules.yml"

# Alerting specifies settings related to the Alertmanager.
alerting:
  alertmanagers:
  - scheme: http
    static_configs:
    - targets:
      - "192.168.1.130:9093"

# remote_write configuration is used by promxy as its local Appender, meaning all
# metrics promxy would "write" (not export) would be sent to this. Examples
# of this include: recording rules, metrics on alerting rules, etc.
remote_write:
  - url: http://victoriametrics:8428/api/v1/write
    queue_config:
      # Number of samples to buffer per shard before we start dropping them.
      capacity: 10000
      # Maximum number of shards, i.e. amount of concurrency.
      max_shards: 1
      # Maximum number of samples per send.
      max_samples_per_send: 500

#remote_write:
  #- url: http://localhost:8083/receive

##
### Promxy configuration
##
promxy:
  server_groups:
    # All upstream prometheus service discovery mechanisms are supported with the same
    # markup, all defined in https://github.com/prometheus/prometheus/blob/master/discovery/config/config.go#L33
    - static_configs:
      - targets:
        - victoriametrics:8428
        # labels to be added to metrics retrieved from this server_group
        labels:
          sg: vm_v1_8428
        anti_affinity: 10s
        remote_read: false
        query_params:
          nocache: 1
        http_client:
          dial_timeout: 1s
          tls_config:
            insecure_skip_verify: true

      # relative_time_range defines a relative-to-now time range that this server group contains.
      # this is completely optional and start/end are both optional as well
      # an example is if this servergroup only has the most recent 3h of data
      # the "start" would be -3h and the end would be left out
      relative_time_range:
        start: -3h
        end: -1h
        truncate: false

      # absolute_time_range defines an absolute time range that this server group contains.
      # this is completely optional and start/end are both optional as well
      # and example is if the servergroup has been deprecated and is no longer receiving data
      # you could set the specific times that it has data for.
      absolute_time_range:
        start: '2009-10-10T23:00:00Z'
        end: '2009-10-11T23:00:00Z'
        truncate: true
```
Prometheus configuration:
```
global:
  scrape_interval:     15s # Set the scrape interval to every 15 seconds. Default is every 1 minute.
  evaluation_interval: 15s # Evaluate rules every 15 seconds. The default is every 1 minute.
  # scrape_timeout is set to the global default (10s).

# Alerting specifies settings related to the Alertmanager.
#alerting:
#  alertmanagers:
#  - scheme: http
#    static_configs:
#    - targets:
#      - "192.168.1.130:9093"
#
rule_files:
  - "/etc/prometheus/rules/linux_system.yml"
  - "/etc/prometheus/rules/windows_system.yml"
  - "/etc/prometheus/rules/windows_process.yml"
  - "/etc/prometheus/rules/docker_container.yml"
  - "/etc/prometheus/rules/process_exporter.yml"
  # - "second_rules.yml"

# remote_write configuration is used by promxy as its local Appender, meaning all
# metrics promxy would "write" (not export) would be sent to this. Examples
# of this include: recording rules, metrics on alerting rules, etc.
remote_write:
  - url: http://victoriametrics:8428/api/v1/write
    queue_config:
      # Number of samples to buffer per shard before we start dropping them.
      capacity: 10000
      # Maximum number of shards, i.e. amount of concurrency.
      max_shards: 1
      # Maximum number of samples per send.
      max_samples_per_send: 500

# A scrape configuration containing exactly one endpoint to scrape:
# Here it's Prometheus itself.
scrape_configs:
  # The job name is added as a label `job=<job_name>` to any timeseries scraped from this config.
  - job_name: 'vmagent'
    static_configs:
      - targets: ['vmagent:8429']
  - job_name: 'victoriametrics'
    static_configs:
      - targets: ['victoriametrics:8428']
  - job_name: 'prometheus'
    # metrics_path defaults to '/metrics'
    # scheme defaults to 'http'.
    static_configs:
    - targets: ['localhost:9090']
  - job_name: 'linux_agent'
    static_configs:
    - targets:
      - '192.168.1.130:9100'
  - job_name: 'windows_agent'
    static_configs:
    - targets:
      - '192.168.10.180:9182'
      labels:
        ip: 192.168.10.180
    relabel_configs:
    - source_labels: ["ip"]
      target_label: "ip"

  - job_name: 'docker'
    static_configs:
    - targets: ['192.168.1.130:9080']
      #labels:
        #instance: docker测试
```

**Comment by ktpktr0** [comment]
The latest test results, the URL in the sample file: http://localhost : 8083 / receive is incorrect, the correct port is 8082.

1. When Prometheus' remote_ When write is configured as Victoria metrics, the proxy server_ Groups are configured as Victoria metrics address. The background of promx returns null results for any query; No alerts are generated for configuring rules on proxy.

2. When Prometheus uses local TSDB, the proxy server_ Groups are configured with two Prometheus addresses, and the background of promx returns to normal for any query; When rules is configured for proxy, the alarm status is normal. The remote_ When write is configured as Victoria metrics, the alarm is also normal.

**Comment by jacksontj** [comment]
This seems like a duplicate of https://github.com/jacksontj/promxy/issues/446 -- TLDR; the targets page in promxy will always be empty (as promxy doesn't scrape metrics from anything directly). This is unfortunately an upstream prom-ism that is a pain to fix (details in that issue).

I think that was the only question here; so I'll close this out; but if there was anything else feel free to re-open or create a new issue :)
